src/Symfony/Component/Workflow/Definition.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow;
+
+use Symfony\Component\Workflow\Exception\LogicException;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class Definition
+{
+    private $places = array();
+
+    private $transitions = array();
+
+    private $initialPlace;
+
+    /**
+     * Definition constructor.
+     *
+     * @param string[]     $places
+     * @param Transition[] $transitions
+     */
+    public function __construct(array $places = array(), array $transitions = array())
+    {
+        $this->addPlaces($places);
+        $this->addTransitions($transitions);
+    }
+
+    public function getInitialPlace()
+    {
+        return $this->initialPlace;
+    }
+
+    public function getPlaces()
+    {
+        return $this->places;
+    }
+
+    public function getTransitions()
+    {
+        return $this->transitions;
+    }
+
+    public function setInitialPlace($place)
+    {
+        if (!isset($this->places[$place])) {
+            throw new LogicException(sprintf('Place "%s" cannot be the initial place as it does not exist.', $place));
+        }
+
+        $this->initialPlace = $place;
+    }
+
+    public function addPlace($place)
+    {
+        if (!preg_match('{^[\w\d_-]+$}', $place)) {
+            throw new InvalidArgumentException(sprintf('The place "%s" contains invalid characters.', $name));
```

Comment:
@gnugat @lyrixx is it fix ? because, in my project it's not working today.

my class is App\Customer and i have an InvalidArgumentException

It doesn't like the backslash in the namespace